modules/AOW_Actions/actions/actionComputeField.php

```diff
@@ -236,7 +236,27 @@ private function resolveRelationParameters(
                     continue;
                 }
 
-                $entity = BeanFactory::getBean($relateFields[$relationParameters[$i]]['module'], $relatedEntityId);
+                if (is_object($relatedEntityId)) {
+                    // If this is a Link2 object then need to use the relationship
+                    // - because it's a one to many relationship's 'one' side
+                    $relationship = $relateFields[$relationParameters[$i]]['link'];
+                    if ($bean->load_relationship($relationship)) {
+                        foreach ($bean->$relationship->getBeans() as $relatedEntity) {
+                            $entity = $relatedEntity;
+                            break;
+                        }
+                    }
+                } elseif (is_string($relatedEntityId)) {
```

Comment:
You can check for valid IDs with a function called `isValidID(...)` in utils.php

Reply:
Thanks for the heads up!